### DISCUSSION TITLE
Copilot stops working on `gender` related subjects

### DISCUSSION THREAD

**Discussion by jebarjonet** [community/community#72603] [Copilot Conversations]
As some people already mentioned [here](https://github.com/orgs/community/discussions/20273) or [here](https://github.com/orgs/community/discussions/55478), **Copilot purposely stops working** on code containing hardcoded banned words from Github such as `gender` or `sex`.

I am labelling this as a **bug** because this behavior is unexpected and undocumented.

I guess you might be embarrassed by what your AI says when it autocompletes gender related topics. However disabling autocompletion is not a great solution since gender can be very present in :
- medical, administrative or demographics studies
- translations in many languages (not much in English though), such as latine languages where nouns do have genders
- sentences generation (for documentation, games, etc.) in those languages

For which Copilot shutting down on most of the files is deceptive and pretty annoying.

Your competitors don't have this problem, which means solutions exist. I hope this will be taken into account someday.

**Comment by RVFET** [comment]
murica is a failed society...

**Comment by marc-pingtt** [comment]
This is ridiculous.

I literally have all the genders written out in the same file so there isn't even 'assumption', and it refuses to work. It is infuriating.

![image](https://github.com/community/community/assets/104793803/e2b40b17-d0cc-4a55-a846-1310008afba9)

**Comment by shanestover8991** [comment]
Good luck deugging! Sometimes you can add an x_ at the front or something to get around bugs like that. Or create codewords/ encryptions basically.

**Reply by Fabizocker456** [reply]
Gondor calls for aid-

**Comment by benjaminjamesbush** [comment]
Wow, that's awful. Devs, please fix this!

**Comment by wsficke** [comment]
Also frustrated by this issue.  Working in the fashion industry, demographic classifications like age and gender are integral to the domain model and entirely apolitical.  This idea of "banned words" is essentially biased - an American bias.  It is long established that ban-list content filtering is terrible.  This artifact of the propagandized American political environment isn't necessary.

Funding schools and improving public literacy is a better way to keep AI-generated propaganda in its place than naive measures like this...

**Reply by OdysseusOperator** [reply]
How is it established that ban-list filtering is terrible ?
Specifically when censoring websites, banlist work way better than blacklisting complete websites.

**Reply by snipe** [reply]
Google the Scunthorpe Problem. https://en.wikipedia.org/wiki/Scunthorpe_problem

**Comment by davidcarvalho83** [comment]
This is idiot. Technology should not be mixed with ideologies. We don't care about this type of things when we are working, we should not!
Stop wanting to shape peoples mind in the wrong context! Co pilot should not interfere about this, if I want to write "gun" as a variable, "sex" or "gender" I should be able to! You (co pilot devs) don't know the purpose and you should not care about that!!

**Comment by nhophucrious** [comment]
This behavior is still on today, and it seems like it's not going away anytime soon. So frustrating, and completely unnecessary.

**Comment by YamiOdymel** [comment]
![image](https://github.com/community/community/assets/7308718/b16a5baa-5733-4286-99c0-e2c40749113e)



It also stops with `trans` while I was referring to `Transactions` 💀💀

**Reply by jmiller-jumptrading** [reply]
I have `trans_time` all over my code and CoPilot refuses to talk about it.

**Reply by YamiOdymel** [reply]
Maybe consider this "Who needs GitHub Copilot when you can roll your own AI code assistant at home"
https://www.theregister.com/2024/08/18/self_hosted_github_copilot/

**Comment by mechanoid23** [comment]
I'm working on a project that has fields defined related to Fish Biology. This has been an interesting thread to find.

**Comment by thegoodduck** [comment]
WOW

**Comment by painebenjamin** [comment]
I am bumping this discussion just to add a voice - the hospital I work at is attempting to harmonize how old and new systems store things like sex and gender identity. This is in an effort to model the social complexities of the topic in our database, as health outcomes have been proven to be demonstrably better when doctors honor a patient's preferred name and gender expression. I am completely unable to get Copilot to assist me in this task.

I can, and will, work around the issue - but I must pose this question. If a rule prevents an engineer from improving patient outcomes, is that an acceptable sacrifice in order to curb misuse?

**Reply by kleineluka** [reply]
I have a hard time believing that it's "preventing an engineer from improving outcomes" when it's an optional tool that doesn't do your job on your behalf in the way you want. Copilot or nobody else isn't going to stop you from writing your own code, the way most people do..

**Reply by jtabox** [reply]
"Why do you care the kitchen knife you pay money every month for is dull? YOU are doing the cooking, a dull knife isn't going to stop you from cooking your own food" (insert Copilot monthly sub price here, for better context)

**Reply by kleineluka** [reply]
I think not being able to auto complete anything related to gender, race, or politics is a non-issue in Copilot at the moment. Manually completing one line of code does not take away from all of the other benefits using Copilot brings, does it?

**Reply by timmc** [reply]
Honestly, it's probably for the best if code that deals in sensitive information like that is not written by clippy.

**Reply by kleineluka** [reply]
I personally believe that's another huge reason that Copilot turns off auto completions with certain keywords or topics - GitHub has no clue what half the training data is, and doesn't want sensitive information to auto fill of real people from the training.

**Reply by jtabox** [reply]
Personally I don't mind, but then my code doesn't contain for example genders or similar things, so I don't encounter this issue often at all. But for someone else's code it can be annoying to have it stop completely. And generally you prefer the services you pay for to not annoy you.

Though I gotta say, recently I was working with a program that had `--f*** <something>` as a parameter, and Copilot would just stop working whenever I focused anywhere in the code after this parameter was mentioned. So it was a bit disruptive to wait for it to autocomplete before realizing it refused. I ended up replacing any menions of that parameter to `--duck`, and there weren't any issues.

The thing is, Copilot does something similar with IP addresses, but instead of completely refusing, it just stops temporarily and waits until you write the IP address, and then starts autocompleting right after. Not sure what the reason is, I assume privacy stuff. But it could do something similar with "bad" words, instead of completely refusing to continue.

**Comment by omarfaroke** [comment]
_**Really!**_

I was the one who thought there was a problem with the auto-complete!! 😅


https://github.com/user-attachments/assets/b4e8e1cd-5320-4f2c-8480-922b39fe1fd9

**Comment by BubblegumYeti** [comment]
I thought AI would be the Terminator timeline... but it seems we are in some sort of stand up comedy AI timeline.

**Comment by jschiefer** [comment]
If this is a way to reliably turn off any AI "help", I'd consider this a feature, not a bug. 

Figuring out how to add enough spiciness to turn off Copilot, yet not get fired from your job, may be difficult, though.

**Comment by jeffreycloudlife** [comment]
Not able to replicate this with Powershell, both with the GPT and Claude models. Created a Gender variable and asked Copilot to fill in sample data. Is it language-specific?

**Reply by painebenjamin** [reply]
I believe the difference there is between Copilot Chat and Copilot Autocomplete - the former is the one where you can choose between different models, while the latter is the one that fails on gender topics. Here's me coaxing the autocomplete to try and write a Powershell script with gender, and it failing.

https://github.com/user-attachments/assets/328f37b1-4ae7-4e59-b46b-044955ad4209

**Comment by Mubelotix** [comment]
I had this issue where it would not even complete a sentence containing "fuck spez"

**Comment by ArmandBernard** [comment]
Also unable to replicate this with the free tier of Copilot for VSCode.

![image](https://github.com/user-attachments/assets/f5cf966d-9b90-44fe-abcd-0ec2180be41b)

![image](https://github.com/user-attachments/assets/916b12a4-cddc-41cb-99bf-51dcb5250fb3)


Maybe this is no longer the case or is region locked?

**Comment by AlpSha** [comment]
I'm still experiencing this on Webstorm while working on a Next.js project. Also happens on Intellij Idea while developing Flutter apps.

https://github.com/user-attachments/assets/756b48bd-5241-4ec7-9de8-2923fca76d31

**Comment by DeveloppeurPascal** [comment]
It's not a bug, it's the best feature ever if we don't want these IA things all over our projects. If adding "trans" ou "gay" in a comment is enough to stop them, see it as a power off button and use it everywhere !

**Comment by bomberstudios** [comment]
Someone asked on a previous comment if this might be location-dependent. I tried this from a non-US country, and Copilot did not disappoint me:

<img width="1056" alt="image" src="https://github.com/user-attachments/assets/af81e3c7-76d6-4b1c-904d-04d9a10278bd" />

**Reply by AlpSha** [reply]
Which country did you try on may I ask? I also use on a non-US country but it does its censor thing.

**Reply by bomberstudios** [reply]
Spain. Not sure if it's relevant but I'm using Copilot through a company license, not the free plan

**Comment by jebarjonet** [comment]
I just tried again Copilot on VSCode and it seems to work again for `gender` and `sex` for me too ! If I write `gender: "m` it autocompletes as `gender: "male"` instead of stopping and it seems to be a very recent change. That's good news ! If this is the case for more people I will close this thread as "resolved".

**Reply by AlpSha** [reply]
I can verify that  too on Intellij. It started to complete ge to gender and ma to male lol

**Reply by LiamDobbelaere** [reply]
This is not resolved, doing const gender: in typescript doesn't suggest any type

**Reply by AmazingTurtle** [reply]
`const gender = '` does not work
though
`const gender='a` or any other letter works and will complete any sort of word, such as male for m, female for f, any for a, r for robot, etc.
Seems to work, right?

**Reply by AurelianSpodarec** [reply]
lol imagine intentional bugs created by devs for devs xD

**Comment by maelp** [comment]
Loooool America is destroying itself by letting dumby Trump and dumber Musk have control over their thoughts and actions 😂

**Comment by ixtli** [comment]
but guys don't use DeepSeek cause it implements censorship!! China BAD !!

**Comment by neopunisher** [comment]
I work at a DNA Forensics lab writing software to track the cases and evidence we analyze, It is reasonable to expect me to interact with these words and when we say sex we are talking about the sex chromosomes. At least now I know why sometimes I don't get completions but a setting would be better...

**Comment by nerder** [comment]
https://github.com/orgs/community/discussions/79223

**Comment by acidvegas** [comment]
lol @ github trying to be woke.....

**Comment by AlpSha** [comment]
I just found out that it doesn't autocomplete the word "immigration" too. There are weirdly banned words. It completes to "Imm" first. Then after typing that part it completes the rest of the word which is "igration". It consistently does it both on dart files and arb file.

![image](https://github.com/user-attachments/assets/06498f05-5765-4e8a-8af3-07ae5bb0f6eb)

**Comment by chrgeor** [comment]
This is ridiculous. It stops working on `y.id` (probably because of https://en.wikipedia.org/wiki/Yid)

**Reply by Fabizocker456** [reply]
Does it still if it's a part of something? (item.cherry.id)?

**Comment by stuartlynne** [comment]
February 2025, Copilot completion (in neovim doing python programming) and Copilot simply refuses to autocomplete gender =

**Comment by Ismael-VC** [comment]
Artificial Stupidity

**Reply by scheeseman486** [reply]
Nothing artificial about it, this is clearly the result of human stupidity.

**Reply by thegoodduck** [reply]
Donald Trump be like:

**Comment by mariavilaro** [comment]
I have to write each one of this "trans" (short for "translation") methods by hand because copilot refuses to autocomplete it. This is beyond ridiculous.

![image](https://github.com/user-attachments/assets/d5c4d8d2-d83c-45a3-87a4-87811ff2628e)

**Reply by Ismael-VC** [reply]
What editor are you using? You can use multiple cursors to write `trans` in all the lines at once, most code editors support this. 
![](https://www.nobledesktop.com/nl/image/ct-multiple-cursors-arrow-keys.gif)

**Comment by AurelianSpodarec** [comment]
Simple solutions.

Stop giving them money and use a competitor - that'll teach them a lesson.

This is the only realistic way you can actually make the change, fix the bug - is to go elsewhere, untill they are forced to fix this bug.

**Comment by artur-barseghyan-springernature** [comment]
Thumbs down for this, GitHub.

**Comment by Diti** [comment]
Any news on when this issue will be resolved?

**Comment by Akash1134** [comment]
Hi everyone,

Thank you for continuing to share your feedback and report issues — it really helps us improve. 🙏 

As part of our ongoing efforts to enhance the experience around content filtering and responsible AI, we’ve put together a new resource:

#### 📘 [GitHub Copilot Responsible AI: Frequently Asked Questions](https://github.com/orgs/community/discussions/164807)

This FAQ addresses common questions and provides guidance on how we're approaching responsible AI with Copilot.

We’d love for you to continue sharing your experiences. If you're encountering issues or have suggestions, please use the information and links in the FAQ to help us support you better.

Regards,
Akash

**Comment by AlpSha** [comment]
Hi @Akash1134,

Let me be honest — the way content filtering is handled in GitHub Copilot is infuriating. You're blocking developers from writing perfectly valid code in 2025 — while the entire internet remains wide open and uncensored. What's the point? Do you really think someone with bad intent needs Copilot to cause harm?

Blocking the word gender, for example, halts completions mid-thought. In what world is that helping developers? This isn't protecting anyone — it's breaking the tool for everyone else.

You say you're working on responsible AI. But this isn’t responsibility — it’s overreach. It’s treating everyone like a potential threat instead of trusting professionals to use a tool as intended: to build, learn, and move fast.

It’s honestly disappointing to see this much time and engineering effort being spent on censorship, while so many real issues and improvements are left untouched.

And if you insist on doubling down on this path, let's be clear — it’s not going to create some "Responsible AI workspace." What it will do is push developers away. Just like I left Copilot for better alternatives. You're not fostering safety — you're alienating the very people who made Copilot successful in the first place.

It's honestly irritating that something as ridiculous as "Responsible AI" is even being used to justify this. We're not trying to misuse your tool. We're just trying to code.

Let us code.

**Note**: This message was drafted with the help of ChatGPT (a responsible AI) to help keep my frustration in check and avoid going too far in expressing just how absurd and counterproductive this situation really is.

**Note 2**: You can’t make a weapon “responsible.” Responsibility lies with the one who uses it.

**Comment by jtabox** [comment]
I'm sorry to be so blunt, but please, I'm not sure if it's your PR department or some Christian Ethics Committee or your bad conscience: What  exactly is the point with those idiotic decisions? Why implement censorship in Copilot and hinder even minor parts in its functionality? What responsible AI are we talking about, can you stop wrapping everything with a stupid protective plastic wrap?

### **Copilot Is Not A Chat Bot**

The developer isn't going to "trick" it into saying s*!t or fv*k or revealing the recipe for MDMA, and then write an article about how their innocence was taken away by evil Copilot.

#### **It's a text completion tool**...

**I'm** the one guiding it wherever **I** want it to go. If for some reason I want it to autocomplete `trans...` then I do not give a singular flying duck if it mistakenly suggests `transgender` (I seriously hope this isn't some kind of trigger word btw). Likewise, I literally couldn't care less that someone in a big corporation's PR department decided that our calculators will have to be "responsible" from now on and will stop working if they are forced to display `666` as the result of a calculation.

If I guide Copilot towards autocompleting something, I probably have a reason and that's that, I'm not interested in any kind of moral hand-holding. If I'm still at elementary school mentally and my reason is that I want to trick Copilot to autocomplete a bad word, no one's going to take me seriously if I start crying online that I guided my autocompletion bot to write a bad word by... writing the first letters of the bad word?

I'm not really sure what imaginary PR-catastrophe you're trying to avoid, but by trying to avoid some unlikely fringe case, **you're actively making your product worse for some of your paying customers.**

Honestly I don't understand the decisions you've been taking for Copilot lately, they're so baffling. and your justifications for them feel at best like completely hollow PR statements, and at worst like sad attempts in gaslighting.

I've had a lot of fun with Copilot, been supporting it financially since before day 1 (was in beta), and defending it whenever it was being unjustly attacked because it was the "mainstream" option... and now I'm at the point where I'm trying to move as far away from it as possible, seriously considering which of the alternatives to choose and have even accepted the fact that I'll need to pay up more each month.

**Reply by AlpSha** [reply]
I resisted so hard not to use the word `stupid` on my above comment. It seems I'm not alone. Here we go...

**Comment by mourad1081** [comment]
2026, this major bug is still ongoing. Please do something about it. I manage multiple projects where patient's gender is a meaningful data. USA, please stop invading the rest of the world's opinions with your stupid opinions. We're dealing with health/governemental/etc data here, not your bulls*t ideologies. We use variable names and properties like gender, sexe, trans (for translation or transaction), dead, cause_suicide, crime names, etc. That's our JOB AND WE'RE PAYING FOR YOUR AI TO WORK PROPERLY.
